### DISCUSSION TITLE
How to view query Supabase receives from Next.js

### DISCUSSION THREAD

**Discussion by richleach** [supabase/supabase#11158] [Questions]
Hi
In my Next.js code I'm trying to pass a query to my Supabase backend and instead of results I get nothing returned (no errors either). I've tested other queries which return expected results, so connection/authentication/config is correct.

I can manually (not dynamically!) piece together the entire query string in my component and send that to my Supabase backend and I get my expected result. However when I attempt to send that same query string to Supabase dynamically I don't get anything (like I said, not even an error message just an empty result set). This tells me my dynamically generated query string is not correct but I can't see what I'm actually sending to Supabase, there's no way for me to see the actual query string THAT WAS SENT.

Does anyone know how I can view the actual query that my Next.js app is sending to my Supabase backend? 

My code issue is that I'm trying to capture the url value from my current page and send that to Supabase in my query:

//incomplete code, just enough to demonstrate what I've done so far....
  const router = useRouter()
  const pathToQ = router.asPath

  const queryToSend = `supabase.from('categories').select('id, url, name').eq('url', '${pathToQ}')`//problem is here!
  console.log(queryToSend);

  useEffect(() => {
    const getData = async () => {
      //const { data, error} = await supabase.from('categories').select(`id, url, name`).eq('url', `'${pathToQ}'`) //returns nothing
      const { data, error} = await supabase.from('categories').select('id, url, name').eq('url', '/blog/code') //returns expected results
      }
    getData();
    
  }, [])

**Comment by GaryAustin1** [comment]
You should be able to see what your code generated by looking at the API Edge logs in the Supabase Dashboard.
Your symptom is also very common of not meeting RLS because your user token is not set up or passed properly.

**Reply by richleach** [reply]
Thank you Gary, I'm sure this will be helpful....
When I attempt to run the following query in the Logs Explorer:
edge_logs

I receive the following error message:
Error when tokenizing at line 1 column 1 near "edge_logs" in "[edge_logs]"

Is there a configuration step I've likely missed? I'm logged into my account as the admin when I created the Supabase account....

**Reply by GaryAustin1** [reply]
Sorry I don't use the logs explorer.
![image](https://user-images.githubusercontent.com/54564956/208953223-093c6e4b-1027-46af-87a7-78d075ec8b33.png)

**Reply by richleach** [reply]
Interesting, it seems like my problem is with how I pass my dynamic params to the query:
To test if my credentials were failing/invalid (since my log files are erroring out, see last post on this thread) I ran this alternative query: 
const { data, error} = await supabase.from("categories").select().order('name',{ascending:true});
Which returns accurate, real time values (I can toggle the ascending value and see the response).

So, to my original posted question/problem, if I'm trapping a url value to use in my query, how do I properly format the string to use when I send my query to Supabase from my Next.js application?

Currently I'm using:
  const pathToQ = router.asPath
 // console.log(pathToQ);
 
  useEffect(() => {
    const getData = async () => {
      const { data, error} = await supabase.from('categories').select('id, url, name').eq('url', pathToQ)  //heeyas ya problem
      //setCatData(catData)
      console.log({data, error});
      setLoading(false)
    }
    getData();
    
  }, [])

The above code says a query is returned but without errors or any records, so I'm confident my formatting of my trapped url value is the culprit. 
Does anyone use a third party logging app? I'm going to need to see my logs if I'm to continue using Supabase as my platform....
Thanks again Gary!
Rich

**Reply by GaryAustin1** [reply]
I don't use next.js, but you have a console.log there commented out is that returning what you expect?
Also there is this note in the next/router docs (don't know if it applies or not):

> Using the asPath field may lead to a mismatch between client and server if the page is rendered using server-side rendering or [automatic static optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization). Avoid using asPath until the isReady field is true.
